test/server/cards/02_SHD/upgrades/Foundling.spec.js

```diff
@@ -0,0 +1,32 @@
+describe('Foundling', function () {
+    integration(function () {
+        describe('Foundling', function () {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['foundling'],
+                        groundArena: ['battlefield-marine', 'pyke-sentinel'],
+                    },
+                    player2: {
+                        groundArena: ['snowspeeder']
+                    }
+                });
+            });
+
+            it('should gain mandalorian\'s trait', function () {
```

Comment:
nit: right now the name of this test will be `Foundling - integration - Foundling should gain mandalorian's trait`. Could we change here and line 3 so that this part becomes something like "Foundling's ability should give the attached card the Mandalorian trait"